editors/vscode/src/client.ts

```diff
@@ -400,41 +400,53 @@ function getGoalsEnvContent(goals : Goal[]){
     let codeHyps : String = ""; //hypothesis HTML code
     let codeGoals : String = ""; //goals HTML code
     let codeEnvGoals : String = ""; //result code HTML
-
+    let goalNo = 0;
+    
     for(let i=0; i < goals.length; i++) {
+        
+        // check if this is a Type Goal
+        if (goals[i].typeofgoal == "Typ "){
```

Comment:
Thanks to add in a comment that the space is not a typo and that this is defined in lsp_base.ml.